<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>mainWindow</class>
 <widget class="QMainWindow" name="mainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1055</width>
    <height>697</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>mainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout_2">
    <item>
     <layout class="QVBoxLayout" name="Center">
      <item>
       <layout class="QHBoxLayout" name="ImageLayout">
        <item>
         <widget class="QGraphicsView" name="GraphicLeft"/>
        </item>
        <item>
         <widget class="QGraphicsView" name="GraphicRight"/>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="SettingLayout">
        <item>
         <layout class="QVBoxLayout" name="LeftLayout">
          <item>
           <widget class="QGroupBox" name="groupBox">
            <property name="title">
             <string>操作参数</string>
            </property>
            <layout class="QVBoxLayout" name="verticalLayout">
             <item>
              <widget class="QSpinBox" name="SpinBoxInput"/>
             </item>
            </layout>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QVBoxLayout" name="RightLayout">
          <item alignment="Qt::AlignVCenter">
           <widget class="QPushButton" name="Generate">
            <property name="text">
             <string>生成</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1055</width>
     <height>22</height>
    </rect>
   </property>
   <widget class="QMenu" name="FileMenu">
    <property name="title">
     <string>文件</string>
    </property>
    <addaction name="actionOpenFile"/>
    <addaction name="actionSave"/>
    <addaction name="actionSaveOther"/>
    <addaction name="actionQuit"/>
   </widget>
   <widget class="QMenu" name="BasedMenu">
    <property name="title">
     <string>基本操作</string>
    </property>
    <addaction name="actionRotate"/>
    <addaction name="actionFlip_Horizontal"/>
    <addaction name="actionFlip_Vertical"/>
    <addaction name="actionScale"/>
    <addaction name="actionGray"/>
   </widget>
   <widget class="QMenu" name="SegmentationMenu">
    <property name="title">
     <string>图像分割</string>
    </property>
    <addaction name="actionEdgeDetection"/>
    <addaction name="actionLineDetection"/>
    <addaction name="actionThresholdProcess"/>
   </widget>
   <widget class="QMenu" name="FrequencyMenu">
    <property name="title">
     <string>频率域算法</string>
    </property>
    <addaction name="actionFourier"/>
    <addaction name="actionLowpassFilter"/>
   </widget>
   <widget class="QMenu" name="DimensionMenu">
    <property name="title">
     <string>空间域算法</string>
    </property>
    <widget class="QMenu" name="SmoothingMenu">
     <property name="title">
      <string>平滑滤波器</string>
     </property>
     <addaction name="actionMedianFilter"/>
     <addaction name="actionGaussFilter"/>
    </widget>
    <widget class="QMenu" name="SharpenMenu">
     <property name="title">
      <string>锐化滤波器</string>
     </property>
     <addaction name="actionLaplacianFilter"/>
     <addaction name="actionSobelFilter"/>
    </widget>
    <addaction name="actionHistogramEqualization"/>
    <addaction name="SmoothingMenu"/>
    <addaction name="SharpenMenu"/>
   </widget>
   <widget class="QMenu" name="MorphologyMenu">
    <property name="title">
     <string>形态学操作</string>
    </property>
    <addaction name="actionDilation"/>
    <addaction name="actionErosion"/>
    <addaction name="actionOpening"/>
    <addaction name="actionClosing"/>
   </widget>
   <widget class="QMenu" name="ArtsMenu">
    <property name="title">
     <string>艺术效果</string>
    </property>
    <addaction name="actionColorPainting"/>
    <addaction name="actionOldPhoto"/>
    <addaction name="actionSketch"/>
   </widget>
   <widget class="QMenu" name="LicenseRecognition">
    <property name="title">
     <string>车牌识别</string>
    </property>
   </widget>
   <widget class="QMenu" name="SettingMenu">
    <property name="title">
     <string>编辑</string>
    </property>
    <addaction name="actionRestore"/>
    <addaction name="actionHistogram"/>
    <addaction name="actionAddNoise"/>
   </widget>
   <widget class="QMenu" name="AboutMenu">
    <property name="title">
     <string>关于</string>
    </property>
   </widget>
   <addaction name="FileMenu"/>
   <addaction name="SettingMenu"/>
   <addaction name="BasedMenu"/>
   <addaction name="SegmentationMenu"/>
   <addaction name="FrequencyMenu"/>
   <addaction name="DimensionMenu"/>
   <addaction name="MorphologyMenu"/>
   <addaction name="ArtsMenu"/>
   <addaction name="LicenseRecognition"/>
   <addaction name="AboutMenu"/>
  </widget>
  <widget class="QToolBar" name="toolBar">
   <property name="windowTitle">
    <string>toolBar</string>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="actionOpenFile"/>
   <addaction name="actionRestore"/>
   <addaction name="actionHistogram"/>
   <addaction name="actionAddNoise"/>
  </widget>
  <action name="actionOpenFile">
   <property name="icon">
    <iconset resource="icons/resources.qrc">
     <normaloff>:/icons/folder.png</normaloff>:/icons/folder.png</iconset>
   </property>
   <property name="text">
    <string>打开</string>
   </property>
   <property name="toolTip">
    <string>打开</string>
   </property>
   <property name="menuRole">
    <enum>QAction::ApplicationSpecificRole</enum>
   </property>
  </action>
  <action name="actionSave">
   <property name="icon">
    <iconset resource="icons/resources.qrc">
     <normaloff>:/icons/diskette.png</normaloff>:/icons/diskette.png</iconset>
   </property>
   <property name="text">
    <string>保存</string>
   </property>
   <property name="toolTip">
    <string>save</string>
   </property>
  </action>
  <action name="actionSaveOther">
   <property name="text">
    <string>另存为</string>
   </property>
  </action>
  <action name="actionQuit">
   <property name="icon">
    <iconset resource="icons/resources.qrc">
     <normaloff>:/icons/close.png</normaloff>:/icons/close.png</iconset>
   </property>
   <property name="text">
    <string>退出</string>
   </property>
   <property name="menuRole">
    <enum>QAction::QuitRole</enum>
   </property>
  </action>
  <action name="actionRotate">
   <property name="icon">
    <iconset resource="icons/resources.qrc">
     <normaloff>:/icons/rotate.png</normaloff>:/icons/rotate.png</iconset>
   </property>
   <property name="text">
    <string>旋转</string>
   </property>
  </action>
  <action name="actionFlip_Horizontal">
   <property name="icon">
    <iconset resource="icons/resources.qrc">
     <normaloff>:/icons/flip-h.png</normaloff>:/icons/flip-h.png</iconset>
   </property>
   <property name="text">
    <string>水平翻转</string>
   </property>
  </action>
  <action name="actionFlip_Vertical">
   <property name="icon">
    <iconset resource="icons/resources.qrc">
     <normaloff>:/icons/flip-v.png</normaloff>:/icons/flip-v.png</iconset>
   </property>
   <property name="text">
    <string>垂直翻转</string>
   </property>
  </action>
  <action name="actionScale">
   <property name="icon">
    <iconset resource="icons/resources.qrc">
     <normaloff>:/icons/resize.png</normaloff>:/icons/resize.png</iconset>
   </property>
   <property name="text">
    <string>缩放</string>
   </property>
  </action>
  <action name="actionGray">
   <property name="text">
    <string>灰度化</string>
   </property>
  </action>
  <action name="actionRestore">
   <property name="icon">
    <iconset resource="icons/resources.qrc">
     <normaloff>:/icons/restore.png</normaloff>:/icons/restore.png</iconset>
   </property>
   <property name="text">
    <string>恢复</string>
   </property>
  </action>
  <action name="actionHistogram">
   <property name="icon">
    <iconset resource="icons/resources.qrc">
     <normaloff>:/icons/bar-chart.png</normaloff>:/icons/bar-chart.png</iconset>
   </property>
   <property name="text">
    <string>直方图</string>
   </property>
  </action>
  <action name="actionEdgeDetection">
   <property name="text">
    <string>边缘检测</string>
   </property>
  </action>
  <action name="actionLineDetection">
   <property name="text">
    <string>直线检测</string>
   </property>
  </action>
  <action name="actionThresholdProcess">
   <property name="text">
    <string>阈值处理</string>
   </property>
  </action>
  <action name="actionFourier">
   <property name="text">
    <string>傅里叶变换</string>
   </property>
  </action>
  <action name="actionHistogramEqualization">
   <property name="text">
    <string>直方图均衡</string>
   </property>
  </action>
  <action name="actionDilation">
   <property name="text">
    <string>膨胀</string>
   </property>
  </action>
  <action name="actionErosion">
   <property name="text">
    <string>腐蚀</string>
   </property>
  </action>
  <action name="actionOpening">
   <property name="text">
    <string>开运算</string>
   </property>
  </action>
  <action name="actionClosing">
   <property name="text">
    <string>闭运算</string>
   </property>
  </action>
  <action name="actionAddNoise">
   <property name="icon">
    <iconset resource="icons/resources.qrc">
     <normaloff>:/icons/noise.png</normaloff>:/icons/noise.png</iconset>
   </property>
   <property name="text">
    <string>添加噪声</string>
   </property>
   <property name="toolTip">
    <string>噪声</string>
   </property>
  </action>
  <action name="actionMedianFilter">
   <property name="text">
    <string>中值滤波</string>
   </property>
  </action>
  <action name="actionGaussFilter">
   <property name="text">
    <string>高斯滤波</string>
   </property>
  </action>
  <action name="actionLaplacianFilter">
   <property name="text">
    <string>Laplacian滤波</string>
   </property>
  </action>
  <action name="actionSobelFilter">
   <property name="text">
    <string>Sobel滤波</string>
   </property>
  </action>
  <action name="actionColorPainting">
   <property name="text">
    <string>水彩画</string>
   </property>
  </action>
  <action name="actionOldPhoto">
   <property name="text">
    <string>怀旧</string>
   </property>
   <property name="toolTip">
    <string>怀旧</string>
   </property>
  </action>
  <action name="actionSketch">
   <property name="text">
    <string>素描</string>
   </property>
  </action>
  <action name="actionLowpassFilter">
   <property name="text">
    <string>低通滤波器</string>
   </property>
  </action>
 </widget>
 <resources>
  <include location="icons/resources.qrc"/>
 </resources>
 <connections/>
</ui>
